### PR TITLE
chore(css): Move CSS examples - Learn area tasks / assessments

### PR DESCRIPTION
**Pull request by bsmth**
### Description

This PR moves tasks (Test your skills) in the Learn area from external examples into the content. 

__Removals:__

* Pages do not link to download starting point

__Additions:__

* Marking hints are moved under the examples at `<details>` elements to show/hide solutions
* See also section, (currently links to CSS building blocks)
* Admonition mentions that you can click "Play" to break example out into an editor

__Changes:__

Converts the following macros to live samples:

```
{{EmbedGHLiveSample("css-examples/learn/tasks/box-model/box-models.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/box-model/inline-block.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/cascade/cascade.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/cascade/cascadelayer.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/images/object-fit.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/images/form.html", '100%', 600)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/overflow/overflow-scroll.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/overflow/overflow-hidden.html", '100%', 1200)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/type.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/class-id.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/pseudo.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/combinators.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/attribute-links.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/sizing/height-min-height.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/sizing/percentages.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/sizing/max-width.html", '100%', 1200)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/tables/table.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/backgrounds/backgrounds1.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/backgrounds/backgrounds2.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/values/color.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/values/length.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/values/position.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/writing-modes/writing-mode.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/writing-modes/logical-width-height.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/writing-modes/logical-mbp.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox1.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox2.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox3.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox4.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/float/float1.html", '100%', 900)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/float/float2.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/float/float3.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid1.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid2.html", '100%', 900)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid3.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid4.html", '100%', 2000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/multicol/multicol1.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/multicol/multicol2.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/multicol/multicol3.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/position/position1.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/position/position2.html", '100%', 1300)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/box-model/box-models.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/box-model/mbp.html", '100%', 600)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/box-model/inline-block.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/cascade/cascade.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/cascade/cascadelayer.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/images/object-fit.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/images/form.html", '100%', 600)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/overflow/overflow-scroll.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/overflow/overflow-hidden.html", '100%', 1200)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/type.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/class-id.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/pseudo.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/combinators.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/selectors/attribute-links.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/sizing/height-min-height.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/sizing/percentages.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/sizing/max-width.html", '100%', 1200)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/tables/table.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/backgrounds/backgrounds1.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/backgrounds/backgrounds2.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/values/color.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/values/length.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/values/position.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/writing-modes/writing-mode.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/writing-modes/logical-width-height.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/writing-modes/logical-mbp.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox1.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox2.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox3.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/flexbox/flexbox4.html", '100%', 1100)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/float/float1.html", '100%', 900)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/float/float2.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/float/float3.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid1.html", '100%', 700)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid2.html", '100%', 900)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid3.html", '100%', 800)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/grid/grid4.html", '100%', 2000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/multicol/multicol1.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/multicol/multicol2.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/multicol/multicol3.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/position/position1.html", '100%', 1000)}}
{{EmbedGHLiveSample("css-examples/learn/tasks/position/position2.html", '100%', 1300)}}
```

### Motivation

Motivation is described in the project issue.

### Related issues and pull requests

- [x] https://github.com/mdn/mdn/issues/597

### CODE REVIEW

File: files/en-us/learn/css/css_layout/grid_skills/index.md [hunk 1]
```diff
@@ -9,68 +9,354 @@ page-type: learn-module-assessment
...
 
-- Can you now cause the first item to display on top without changing the order of items in the source?
+.item2 {
+}
+```
 
-> [!CALLOUT]
->
-> [Download the starting point for this task](https://github.com/mdn/css-examples/blob/main/learn/tasks/grid/grid2-download.html) to work in your own editor or in an online editor.
+{{EmbedLiveSample("grid2", "", "300px")}}
```
Comment (chrisdavidmills) [inline files/en-us/learn/css/css_layout/grid_skills/index.md:123]: increase the embed height a little bit to get rid of the scrollbar?

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Nice work, @bsmth. I've left you a few comments here too, but nothing much.

File: files/en-us/learn/css/building_blocks/values_tasks/index.md [hunk 2]
```diff
@@ -62,10 +171,43 @@ Your final result should look like the image below:
...
+```
+
+```css live-sample___position
+.box {
+  background-image: url(https://mdn.github.io/shared-assets/images/examples/star-pink_256x256.png);
+  background-repeat: no-repeat;
+}
+```
+
+{{EmbedLiveSample("position", "", "400px")}}
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/learn/css/building_blocks/values_tasks/index.md:194]: True, that was a placeholder until this landed:

- [x] https://github.com/mdn/shared-assets/pull/31

It's available now, so I'll replace

[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/learn/css/building_blocks/values_tasks/index.md:194]: Fixed in cd8709278f9f50b9c2a74a7f201483c7cc48870c

File: files/en-us/learn/css/building_blocks/tables_tasks/index.md [hunk 3]
```diff
@@ -31,14 +31,145 @@ Our finished table will look like the image below. There are a number of ways th
...
+}
+
+tfoot tr :nth-child(2) {
+  text-align: left;
+}
+```
+
+</details>
+
+**Bonus question:** What can you do to make the table layout behave a bit more predictably? Think of how table columns are sized by default and how we can change this behavior to size the columns according to the width of their headings.
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/learn/css/building_blocks/tables_tasks/index.md]: done in 620042cdded96e692bd5a372feec6564cc233fe8

File: files/en-us/learn/css/building_blocks/selectors/selectors_tasks/index.md [hunk 4]
```diff
@@ -105,10 +336,72 @@ Your final result should look like the image below:
...
+```
+
+```css live-sample___attribute-links
+a {
+  border: 5px solid grey;
+}
+/* Add styles here */
+```
+
+{{EmbedLiveSample("attribute-links", "", "250px")}}
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/learn/css/building_blocks/selectors/selectors_tasks/index.md]: done in b20ef682d40b2e2cd1d5d455e8c79972785248ad

File: files/en-us/learn/css/css_layout/grid_skills/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/learn/css/css_layout/grid_skills/index.md]: done in 31d4d80dfa

**Comment by bsmth** [comment]
Thank you! Should be ready for another once-over 🔬

**Review by chrisdavidmills — APPROVED** [review]
LGTM! Merging!

**Comment by bsmth** [comment]
Thank you!

**Pull request merged**